File: ChatSupport/src/main/java/com/vantagecircle/chatapp/adapter/GroupMAdapter.java
package com.vantagecircle.chatapp.adapter;

import com.firebase.ui.database.FirebaseRecyclerAdapter;
import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.Query;
import com.vantagecircle.chatapp.Support;
import com.vantagecircle.chatapp.holder.GroupMViewHolder;
import com.vantagecircle.chatapp.interfacePref.ClickGroup;
import com.vantagecircle.chatapp.model.GroupM;

/**
 * Created by bapidas on 21/07/17.
 */

public class GroupMAdapter extends FirebaseRecyclerAdapter<GroupM, GroupMViewHolder> {
    private ClickGroup clickGroup;

    public GroupMAdapter(Class<GroupM> modelClass, int modelLayout, Class<GroupMViewHolder> viewHolderClass,
                         Query ref, ClickGroup clickGroup) {
        super(modelClass, modelLayout, viewHolderClass, ref);
        this.clickGroup = clickGroup;
    }

    public GroupMAdapter(Class<GroupM> modelClass, int modelLayout, Class<GroupMViewHolder> viewHolderClass,
                         DatabaseReference ref, ClickGroup clickGroup) {
        super(modelClass, modelLayout, viewHolderClass, ref);
        this.clickGroup = clickGroup;
    }

    @Override
    protected GroupM parseSnapshot(DataSnapshot snapshot) {
        if(snapshot.child("users").hasChild(Support.id)){
            return super.parseSnapshot(snapshot);
        }
        return null;
    }

    @Override
    protected void populateViewHolder(GroupMViewHolder viewHolder, GroupM model, int position) {
        viewHolder.setHolderData(model, clickGroup);
    }
}

File: app/src/main/java/com/bapidas/chattingapp/ui/adapter/GroupMAdapter.kt
package com.bapidas.chattingapp.ui.adapter

import com.bapidas.chattingapp.ChatApplication
import com.bapidas.chattingapp.data.model.GroupM
import com.bapidas.chattingapp.ui.adapter.callbacks.ClickGroup
import com.bapidas.chattingapp.ui.adapter.holder.GroupMViewHolder
import com.firebase.ui.database.FirebaseRecyclerAdapter
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.Query

/**
 * Created by bapidas on 21/07/17.
 */
class GroupMAdapter(modelClass: Class<GroupM>, modelLayout: Int,
                    viewHolderClass: Class<GroupMViewHolder>,
                    ref: Query?,
                    private var clickGroup: ClickGroup) :
        FirebaseRecyclerAdapter<GroupM, GroupMViewHolder>(modelClass, modelLayout, viewHolderClass, ref) {

    override fun parseSnapshot(snapshot: DataSnapshot): GroupM? {
        return if (snapshot.child("users")
                        .hasChild(ChatApplication.applicationContext().id.orEmpty())) {
            super.parseSnapshot(snapshot)
        } else null
    }

    override fun populateViewHolder(viewHolder: GroupMViewHolder, model: GroupM?, position: Int) {
        model?.let { viewHolder.setHolderData(it, clickGroup) }
    }
}